src/utils/init/config-github.js

```diff
@@ -1,188 +1,215 @@
+      message = `${message}. Does the repository ${name} exist and do ${owner} has the correct permissions to set up deploy keys?`
+    }
+    failAndExit(message)
   }
-  if (build && build.functions) console.log(`Netlify functions folder is ${chalk.yellow(build.functions)}`)
-  const { buildCmd, buildDir } = await inquirer.prompt([
-    {
-      type: 'input',
-      name: 'buildCmd',
-      message: 'Your build command (hugo build/yarn run build/etc):',
-      filter: (val) => (val === '' ? '# no build command' : val),
-      default: defaultBuildCmd,
-    },
-    {
-      type: 'input',
-      name: 'buildDir',
-      message: 'Directory to deploy (blank for current dir):',
-      default: defaultBuildDir,
-    },
-  ])
+}
 
-  const tomlpath = path.join(ctx.netlify.site.root, 'netlify.toml')
-  const tomlDoesNotExist = !fs.existsSync(tomlpath)
-  if (tomlDoesNotExist && (!ctx.netlify.config || Object.keys(ctx.netlify.config).length === 0)) {
-    const { makeNetlifyTOML } = await inquirer.prompt([
-      {
-        type: 'confirm',
-        name: 'makeNetlifyTOML',
-        message: 'No netlify.toml detected. Would you like to create one with these build settings?',
-        default: true,
-      },
-    ])
-    if (makeNetlifyTOML && ctx.netlify.site && ctx.netlify.site.root) {
-      fs.writeFileSync(tomlpath, makeNetlifyTOMLtemplate({ command: buildCmd, publish: buildDir }))
-    } else {
-      throw new Error('NetlifyCLIError: expected there to be a Netlify site root, please investigate', ctx.netlify.site)
+const getGitHubRepo = async ({ octokit, repoOwner, repoName, failAndExit }) => {
+  try {
+    const { data } = await octokit.repos.get({
+      owner: repoOwner,
+      repo: repoName,
+    })
+    return data
+  } catch (error) {
+    let message = formatErrorMessage({ message: 'Failed retrieving GitHub repository information', error })
+    if (error.status === 404) {
+      const { name, owner } = formatRepoAndOwner({ repoName, repoOwner })
+      message = `${message}. Does the repository ${name} exist and accessible by ${owner}`
     }
+    failAndExit(message)
   }
+}
 
-  repo.dir = buildDir
-
-  if (buildCmd) {
-    repo.cmd = buildCmd
+const hookExists = async ({ deployHook, octokit, repoOwner, repoName }) => {
+  try {
+    const { data: hooks } = await octokit.repos.listHooks({
+      owner: repoOwner,
+      repo: repoName,
+      per_page: PAGE_SIZE,
+    })
+    const exists = hooks.some((hook) => hook.config.url === deployHook)
+    return exists
+  } catch (_) {
+    // we don't need to fail if listHooks errors out
+    return false
   }
+}
 
-  const results = await octokit.repos.get({
-    owner: parsedURL.owner,
-    repo: parsedURL.name,
-  })
-
-  repo.id = results.data.id
-  repo.repo_path = results.data.full_name
-  repo.repo_branch = results.data.default_branch
-  repo.allowed_branches = [results.data.default_branch]
-
-  site = await api.updateSite({ siteId: site.id, body: { repo } })
```

Comment:
Please note that `repo` is the same one returned from `getRepoData` so a lot of redundant data was sent to our API